js/style/style.js

```diff
@@ -247,19 +251,95 @@ Style.prototype = util.inherit(Evented, {
         zh.lastZoom = z;
     },
 
+    _checkLoaded: function () {
+        if (!this._loaded) {
+            throw new Error('Style is not done loading');
+        }
+    },
+
     /**
-     * Apply multiple style mutations in a batch
-     * @param {function} work Function which accepts the StyleBatch interface
+     * Apply queued style mutations in a batch
      * @private
      */
-    batch: function(work) {
-        styleBatch(this, work);
+    update: function(classes, options) {
```

Comment:
I've renamed `mutations`, `broadcastLayers` and did some other minor cleanups/renames, but leaving the rest for future PRs (especially `cascade` and `recalculate`) so that it doesn't interfere with current DDS PRs, otherwise we would have a lot of conflicts to resolve.